tool/release/rules.bzl

```diff
@@ -16,42 +16,29 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
-RELEASE_VALIDATE_DEPS_SCRIPT_TEMPLATE = """
-import json
-import sys
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
 
-with open("{workspace_refs_json_path}") as f:
-    workspace_refs = json.load(f)
+def _release_validate_deps_script_impl(ctx):
+    test_script = ctx.actions.declare_file("{}.kt".format(ctx.attr.name))
 
-tagged_deps = set("{tagged_deps}".split(','))
-snapshot_dependencies = tagged_deps - set(workspace_refs['tags'])
-
-if snapshot_dependencies:
-    print('These dependencies are excepted to be declared by tag instead of commit: {}'.format(snapshot_dependencies))
-    sys.exit(1)
-"""
-
-def _release_validate_deps_test_impl(ctx):
-    test_script = ctx.actions.declare_file("{}.py".format(ctx.attr.name))
-
-    ctx.actions.write(
+    ctx.actions.expand_template(
         output = test_script,
-        content = RELEASE_VALIDATE_DEPS_SCRIPT_TEMPLATE.format(
-            workspace_refs_json_path=ctx.file.refs.path,
-            tagged_deps=ctx.attr.tagged_deps
-        ),
-        is_executable = True,
+        template = ctx.file._release_validate_deps_template,
+        substitutions = {
+            "{workspace_refs_json_path}": ctx.file.refs.path,
+            "{tagged_deps}": ",".join(ctx.attr.tagged_deps),
+        }
     )
 
     return [
         DefaultInfo(
-            executable = test_script,
-            runfiles = ctx.runfiles(files = [test_script, ctx.file.refs])
+            runfiles = ctx.runfiles(files = [ctx.file.refs]),
+            files = depset(direct = [test_script])
         )
     ]
 
 
-release_validate_deps_test = rule(
+release_validate_deps_script = rule(
```

Comment:
@flyingsilverfin can we prefix it with `_` so it's clear that this rule is not made for public consumption?

Reply:
yep ok